libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2834,15 +2830,22 @@ string YulUtilFunctions::allocationTemporaryMemoryFunction()
 	});
 }
 
-string YulUtilFunctions::releaseTemporaryMemoryFunction()
+string YulUtilFunctions::finalizeAllocationFunction()
 {
-	string functionName = "releaseTemporaryMemory";
-	return m_functionCollector.createFunction(functionName, [&](){
+	string functionName = "finalize_allocation";
+	return m_functionCollector.createFunction(functionName, [&]() {
 		return Whiskers(R"(
-			function <functionName>() {
+			function <functionName>(memPtr, size) {
+				let newFreePtr := add(memPtr, <roundUp>(size))
```

Comment:
Does `size` need to be rounded up even if it is already a multiple of 32?

Reply:
If it is already a multiple of 32, it will not change.

Reply:
Ah I see, thanks!